packages/rpc-subscriptions-transport-websocket/src/websocket-connection.ts

```diff
@@ -24,10 +24,9 @@ type IteratorState =
           onError: Parameters<ConstructorParameters<typeof Promise>[0]>[1];
           onMessage: Parameters<ConstructorParameters<typeof Promise>[0]>[0];
       };
-export type RpcWebSocketConnection = Readonly<{
+export interface RpcWebSocketConnection extends AsyncIterable<unknown> {
     send(payload: unknown): Promise<void>;
-    [Symbol.asyncIterator](): AsyncGenerator<unknown>;
```

Comment:
If you want to know more, check out the TypeScript types for `AsyncGenerator` and `AsyncIterable`.

Basically I don't want the types to indicate that this is possible:

```ts
const notifications = await rpcSubscriptions...subscribe();

const iterator = notifications[Symbol.asyncIterator]()[Symbol.asyncIterator]()[Symbol.asyncIterator]()[Symbol.asyncIterator](); // NO
```